### ISSUE TITLE
HLookup::hLookupSearch(..) sends wrong arguments to LookupBase::checkMatch(..).

### ISSUE THREAD

**Issue by majermi4** [PHPOffice/PhpSpreadsheet#2123]
This is:

```
- [X] a bug report
```

### What is the expected behavior?

Calling `$cell->getCalculatedValue()` (`$cell` is instance of `PhpOffice\PhpSpreadsheet\Cell`) should work but it fails with the latest master.

### What is the current behavior?

After $cell->getCalculatedValue() is called, the program fails with following error:

```
TypeError : PhpOffice\PhpSpreadsheet\Calculation\LookupRef\LookupBase::checkMatch(): Argument #4 ($rowKey) must be of type int, string given, called in 
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php on line 74
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/LookupBase.php:25
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php:74
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php:45
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:4776
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3469
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3262
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/Cell.php:259
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:5192
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:4669
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3469
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3262
 xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/Cell.php:259
```

Here is the $cell variable value before `$cell->getCalculatedValue()` is called:

![Screenshot 2021-05-27 at 17 25 14](https://user-images.githubusercontent.com/2118173/119853546-97143d00-bf10-11eb-8c00-5b73e4bb55f2.png)

And here are the values from a breakpoint right before the error is triggered at
`HLookup.php:73, PhpOffice\PhpSpreadsheet\Calculation\LookupRef\HLookup::hLookupSearch()`

![Screenshot 2021-05-27 at 17 30 04](https://user-images.githubusercontent.com/2118173/119854256-36393480-bf11-11eb-9316-47eca8816e6a.png)

### What are the steps to reproduce?

Unfortunately, I cannot simply share the the excel file that fails because it is from one of our customers but let me know if any more information is needed apart from what was shared above. ;-)

### Which versions of PhpSpreadsheet and PHP are affected?

Tested on current master branch with PHP 8.0 and 7.4.

**Comment by majermi4** [comment]
This is still and issue for us so I created a test excel sheet to demonstrate the problem.

Running this code:
```
$spreadsheet = IOFactory::load('./hlookup-test.xlsx');
$sheet = $spreadsheet->getSheet(0);
$cellVal = $sheet->getCell('H8')->getCalculatedValue();
```

On this excel spreadsheet: [hlookup-test.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6649200/hlookup-test.xlsx)

![Screenshot 2021-06-14 at 16 49 22](https://user-images.githubusercontent.com/2118173/121911997-7d585f80-cd30-11eb-904c-7b848acf91e6.png)

Results in: 
```
TypeError {#3754
  #message: "Argument 4 passed to PhpOffice\PhpSpreadsheet\Calculation\LookupRef\LookupBase::checkMatch() must be of the type int, string given, called in xxx/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php on line 74"
  #code: 0
  #file: "./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/LookupBase.php"
  #line: 25
  trace: {
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/LookupBase.php:25 { …}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php:74 { …}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/LookupRef/HLookup.php:45 { …}
    PhpOffice\PhpSpreadsheet\Calculation\LookupRef\HLookup::lookup() {}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:4846 { …}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3539 { …}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php:3332 { …}
    ./vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Cell/Cell.php:259 { …}
    ./src/MyNamespace/Command/TestExcel.php:30 {
      MyNamespace\Command\TestExcel->execute(InputInterface $input, OutputInterface $output): void
      › $sheet = $spreadsheet->getSheet(0);
      › $cellVal = $sheet->getCell('H8')->getCalculatedValue();
      › 
    }
```

ping @MarkBaker let me know if I should provide any more information ;-)

**Comment by fabioestevesrs** [comment]
Hi!

I had the same problem.
- Error:
Fatal error:  Uncaught TypeError: Argument 4 passed to PhpOffice\PhpSpreadsheet\Calculation\LookupRef\LookupBase::checkMatch() must be of the type int, string given

- Data organized in method extractCellRange (class vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Calculation/Calculation.php):
array(46) {
  [2]=>
  array(7) {
    ["B"]=>
    string(4) "teste"
    ["C"]=>
    string(25) "teste1"
    ["D"]=>
    string(16) "teste1"
  }
}

Also waiting for a solution to the problem. =)

**Comment by PowerKiKi** [comment]
Fixed in [1.19.0](https://github.com/PHPOffice/PhpSpreadsheet/releases/tag/1.19.0)
